dist/vue-a11y-dialog.esm.js

```diff
@@ -0,0 +1,333 @@
+      if (dialog) {
+        dialog.destroy();
+      }
+      emit("dialogRef");
+    });
+    return {
+      dialog,
+      close,
+      portalTarget,
+      fullTitleId,
+      roleAttribute,
+      rootElement
+    };
+  }
+};
+const _hoisted_1 = ["id", "role", "aria-labelledby"];
+const _hoisted_2 = ["aria-label"];
+const _hoisted_3 = /* @__PURE__ */ createTextVNode(/* @__PURE__ */ toDisplayString("\xD7"));
+const _hoisted_4 = ["id"];
+const _hoisted_5 = ["aria-label"];
+const _hoisted_6 = /* @__PURE__ */ createTextVNode(/* @__PURE__ */ toDisplayString("\xD7"));
+function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
+  return openBlock(), createBlock(Teleport, { to: $setup.portalTarget }, [
+    createElementVNode("div", {
+      ref: "rootElement",
+      id: $props.id,
+      class: normalizeClass($props.classNames.container),
+      role: $setup.roleAttribute,
+      "aria-hidden": "true",
+      "aria-labelledby": $setup.fullTitleId
+    }, [
+      createElementVNode("div", {
+        "data-a11y-dialog-hide": "",
+        tabIndex: "-1",
+        class: normalizeClass($props.classNames.overlay),
+        onClick: _cache[0] || (_cache[0] = ($event) => $props.role === "alertdialog" ? void 0 : $setup.close)
+      }, null, 2),
+      createElementVNode("div", {
+        role: "document",
+        class: normalizeClass($props.classNames.document)
+      }, [
+        $props.closeButtonPosition == "first" ? (openBlock(), createElementBlock("button", {
+          key: 0,
+          "data-a11y-dialog-hide": "",
+          type: "button",
+          "aria-label": $props.closeButtonLabel,
+          class: normalizeClass($props.classNames.closeButton),
+          onClick: _cache[1] || (_cache[1] = (...args) => $setup.close && $setup.close(...args))
+        }, [
+          renderSlot(_ctx.$slots, "closeButtonContent", {}, () => [
+            _hoisted_3
+          ])
+        ], 10, _hoisted_2)) : createCommentVNode("", true),
+        createElementVNode("p", {
+          id: $setup.fullTitleId,
+          class: normalizeClass($props.classNames.title)
+        }, [
+          renderSlot(_ctx.$slots, "title")
+        ], 10, _hoisted_4),
+        renderSlot(_ctx.$slots, "default"),
+        $props.closeButtonPosition == "last" ? (openBlock(), createElementBlock("button", {
+          key: 1,
+          "data-a11y-dialog-hide": "",
+          type: "button",
+          "aria-label": $props.closeButtonLabel,
+          class: normalizeClass($props.classNames.closeButton),
+          onClick: _cache[2] || (_cache[2] = (...args) => $setup.close && $setup.close(...args))
+        }, [
+          renderSlot(_ctx.$slots, "closeButtonContent", {}, () => [
+            _hoisted_6
+          ])
+        ], 10, _hoisted_5)) : createCommentVNode("", true)
+      ], 2)
+    ], 10, _hoisted_1)
+  ], 8, ["to"]);
+}
+var A11yDialog = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
+var index = {
+  install: (app) => {
```

Comment:
This is what I saw that made me realize you need to import the default and then literally call `foo.install(app)` on it. If you follow the logic through it makes sense since it's giving us `index as default` and index is just an object literal -- so you have to call `install` yourself. This is the first time I've seen or used this pattern so there's the chance there's something more idiomatic I'm totally unaware of.